ibis/backends/base/sqlglot/compiler.py

```diff
@@ -192,22 +192,181 @@ class SQLGlotCompiler(abc.ABC):
+        ops.Exp: "exp",
+        ops.First: "first",
+        ops.FirstValue: "first_value",
+        ops.GroupConcat: "group_concat",
+        ops.IfElse: "if",
+        ops.IsInf: "isinf",
+        ops.IsNan: "isnan",
+        ops.JSONGetItem: "json_extract",
+        ops.LPad: "lpad",
+        ops.Last: "last",
+        ops.LastValue: "last_value",
+        ops.Levenshtein: "levenshtein",
+        ops.Ln: "ln",
+        ops.Log10: "log",
+        ops.Log2: "log2",
+        ops.Lowercase: "lower",
+        ops.Map: "map",
+        ops.Median: "median",
+        ops.MinRank: "rank",
+        ops.NTile: "ntile",
+        ops.NthValue: "nth_value",
+        ops.NullIf: "nullif",
+        ops.PercentRank: "percent_rank",
+        ops.Pi: "pi",
+        ops.Power: "pow",
+        ops.RPad: "rpad",
+        ops.Radians: "radians",
+        ops.RandomScalar: "random",
+        ops.RegexSearch: "regexp_like",
+        ops.RegexSplit: "regexp_split",
+        ops.Repeat: "repeat",
+        ops.Reverse: "reverse",
+        ops.RowNumber: "row_number",
+        ops.Sign: "sign",
+        ops.Sin: "sin",
+        ops.Sqrt: "sqrt",
+        ops.StartsWith: "starts_with",
+        ops.StrRight: "right",
+        ops.StringAscii: "ascii",
+        ops.StringContains: "contains",
+        ops.StringLength: "length",
+        ops.StringReplace: "replace",
+        ops.StringSplit: "split",
+        ops.StringToTimestamp: "str_to_time",
+        ops.Tan: "tan",
+        ops.Translate: "translate",
+        ops.Unnest: "explode",
+        ops.Uppercase: "upper",
+    }
+
+    BINARY_INFIX_OPS = {
+        # Binary operations
+        ops.Add: sge.Add,
+        ops.Subtract: sge.Sub,
+        ops.Multiply: sge.Mul,
+        ops.Divide: sge.Div,
+        ops.Modulus: sge.Mod,
+        ops.Power: sge.Pow,
+        # Comparisons
+        ops.GreaterEqual: sge.GTE,
+        ops.Greater: sge.GT,
+        ops.LessEqual: sge.LTE,
+        ops.Less: sge.LT,
+        ops.Equals: sge.EQ,
+        ops.NotEquals: sge.NEQ,
+        # Boolean comparisons
+        ops.And: sge.And,
+        ops.Or: sge.Or,
+        ops.Xor: sge.Xor,
+        # Bitwise business
+        ops.BitwiseLeftShift: sge.BitwiseLeftShift,
+        ops.BitwiseRightShift: sge.BitwiseRightShift,
+        ops.BitwiseAnd: sge.BitwiseAnd,
+        ops.BitwiseOr: sge.BitwiseOr,
+        ops.BitwiseXor: sge.BitwiseXor,
+        # Time arithmetic
+        ops.DateAdd: sge.Add,
+        ops.DateSub: sge.Sub,
+        ops.DateDiff: sge.Sub,
+        ops.TimestampAdd: sge.Add,
+        ops.TimestampSub: sge.Sub,
+        ops.TimestampDiff: sge.Sub,
+        # Interval Marginalia
+        ops.IntervalAdd: sge.Add,
+        ops.IntervalMultiply: sge.Mul,
+        ops.IntervalSubtract: sge.Sub,
+    }
+
     def __init__(self) -> None:
         self.agg = AggGen(aggfunc=self._aggregate)
         self.f = FuncGen()
         self.v = VarGen()
 
     def __init_subclass__(cls, **kwargs):
```

Comment:
So the only reason we need to construct and attach methods here is because of `has_operation` looks for the method? Couldn't we move this logic into a handler which would compile all value nodes if those are not explicitly overwritten?

Reply:
Maybe, but I don't want to spend time making this improvement perfect here. If you have an idea for a further improvement, that can be done in a follow up.